plugins/auth-backend/src/identity/StaticKeyStore.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AnyJWK, KeyStore, StoredKey } from './types';
+import { exportJWK, importPKCS8, importSPKI, JWK } from 'jose';
+import { KeyLike } from 'jose/dist/types/types';
+import { promises as fs } from 'fs';
+import { Config } from '@backstage/config';
+
+export type KeyPair = {
+  publicKey: JWK;
+  privateKey: JWK;
+};
+
+export type StaticKeyConfig = {
+  publicKeyFile: string;
+  privateKeyFile: string;
+  keyId: string;
+  algorithm: string;
+};
+
+const DEFAULT_ALGORITHM = 'ES256';
+
+/**
+ * Key store that loads predefined public/private key pairs from disk
+ *
+ * The private key should be represented using the PKCS#8 format,
+ * while the public key should be in the SPKI format.
+ *
+ * @remarks
+ *
+ * You can generate a public and private key pair, using
+ * openssl:
+ *
+ * Generate a private key using the ES256 algorithm
+ * ```sh
+ * openssl ecparam -name prime256v1 -genkey -out private.ec.key
+ * ```
+ * Convert it to PKCS#8 format
+ * ```sh
+ * openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in private.ec.key -out private.key
+ * ```
+ * Extract the public key
+ * ```sh
+ * openssl ec -inform PEM -outform PEM -pubout -in private.key -out public.key
+ * ```
+ *
+ * Provide the paths to private.key and public.key as the respective
+ * private and public key paths in the StaticKeyStore.create(...) method.
+ */
+export class StaticKeyStore implements KeyStore {
+  private readonly keyPairs: KeyPair[];
+  private readonly createdAt: Date;
+
+  private constructor(keyPairs: KeyPair[]) {
+    if (keyPairs.length === 0) {
+      throw new Error('Should provide at least one key pair');
+    }
+
+    this.keyPairs = keyPairs;
+    this.createdAt = new Date();
+  }
+
+  public static async fromConfig(config: Config): Promise<StaticKeyStore> {
+    const keyConfigs = config.getConfigArray('keys').map(c => {
```

Comment:
it's arguably common that the config given to `fromConfig` methods is the root config, not a specific subtree - this makes it easier to construct in isolation without the caller having to know exactly where its config exists